core/trie/bitarray.go

```diff
@@ -0,0 +1,458 @@
+package trie
+
+import (
+	"bytes"
+	"encoding/binary"
+	"encoding/hex"
+	"fmt"
+	"math"
+	"math/bits"
+
+	"github.com/NethermindEth/juno/core/felt"
+)
+
+const (
+	maxUint64 = uint64(math.MaxUint64) // 0xFFFFFFFFFFFFFFFF
+	bits8     = 8
+)
+
+var emptyBitArray = new(BitArray)
+
+// Represents a bit array with length representing the number of used bits.
+// It uses a little endian representation to do bitwise operations of the words efficiently.
+// For example, if len is 10, it means that the 2^9, 2^8, ..., 2^0 bits are used.
+// The max length is 255 bits (uint8), because our use case only need up to 251 bits for a given trie key.
+// Although words can be used to represent 256 bits, we don't want to add an additional byte for the length.
+type BitArray struct {
+	len   uint8     // number of used bits
+	words [4]uint64 // little endian (i.e. words[0] is the least significant)
+}
+
+func NewBitArray(length uint8, val uint64) *BitArray {
+	return new(BitArray).SetUint64(length, val)
+}
```

Comment:
`BitArray` escapes to heap in this case. What do you think of returning it as a value type instead?